### PR TITLE
Bring in apple client into monorepo

### PR DESCRIPTION
**Pull request by roop**
This PR brings in the apple client into the monorepo.

### CODE REVIEW

File: swift/apple/USING_UNRELEASED_CONNLIB.md
Comment (francesca64) [inline swift/apple/USING_UNRELEASED_CONNLIB.md]: It looks like you changed `Package.swift` to use local connlib, and there's `build-xcframework-dev.sh` now, so how much of this guide is still relevant?

**Review by francesca64 — DISMISSED** [review]

File: swift/apple/USING_UNRELEASED_CONNLIB.md
Comment (roop) [reply swift/apple/USING_UNRELEASED_CONNLIB.md]: You're right. These instructions are not valid anymore, so I removed this file in [81021e9](https://github.com/firezone/firezone/pull/1737/commits/81021e94bbaaeb1f3300efcef31b11ebe15f6176)

**Comment by roop** [comment]
The integration test CI failure seems to be unrelated to this change. Not sure if we should even be running it for changes in the `swift/` folder.

File: .github/workflows/swift.yml
Comment (jamilbk) [inline .github/workflows/swift.yml:44]: Probably good to apply the fix from #1734 
```suggestion
          workspaces: ./rust
          prefix-key: rust-${{ matrix.runs-on }}
          save-if: ${{ github.ref == 'refs/heads/cloud' }}
```

File: swift/apple/.github/workflows_disabled/ci.yml
Comment (jamilbk) [inline swift/apple/.github/workflows_disabled/ci.yml:1]: All workflows should be in the top-level `.github`, so you'll probably want to move this to somewhere like `.github/workflows_disabled/swift.yml` to be consistent with the others.

File: swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift
Comment (jamilbk) [inline swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift:69]: Slight improvement over this hack from https://developer.apple.com/forums/thread/84920?page=2

```suggestion
      // Hacky workaround for Apple bug 32073323.
      // See discussion at https://developer.apple.com/forums/thread/84920
      Darwin.usleep(500000)
```

File: swift/apple/LICENSE
Comment (jamilbk) [inline swift/apple/LICENSE]:
```suggestion
   Copyright 2023 Firezone, Inc.
```

File: swift/apple/LICENSE
Comment (jamilbk) [inline swift/apple/LICENSE:1]: Still figuring out how to manage licensing for the monorepo, but this file works for now.

File: swift/apple/Makefile
Comment (jamilbk) [inline swift/apple/Makefile]:
```suggestion
```

File: swift/apple/Makefile
Comment (jamilbk) [inline swift/apple/Makefile]:
```suggestion
		-destination platform="macOS,arch=$(uname -m)"
```

File: swift/apple/PortalMock/data/jwks.json
Comment (jamilbk) [inline swift/apple/PortalMock/data/jwks.json:1]: This file can be deleted

File: swift/apple/PortalMock/data/jwks_priv.json
Comment (jamilbk) [inline swift/apple/PortalMock/data/jwks_priv.json:1]: This file can be deleted.

File: swift/apple/PortalMock/data/jwks.json
Comment (jamilbk) [inline swift/apple/PortalMock/data/jwks.json:1]: This file can be deleted

File: swift/apple/PortalMock/data/client_csrf_token
Comment (jamilbk) [inline swift/apple/PortalMock/data/client_csrf_token:1]: This file can be deleted

**Review by jamilbk — CHANGES_REQUESTED** [review]
Thanks for taking this on. Left a few minor comments/suggestions.

Let's do the Connlib refactor after this gets merged (cc @francesca64)

File: .github/workflows/swift.yml
Comment (roop) [reply .github/workflows/swift.yml:44]: Okay, done. Commit: 7065755

File: swift/apple/.github/workflows_disabled/ci.yml
Comment (roop) [reply swift/apple/.github/workflows_disabled/ci.yml:1]: I removed it. I don't think we need it.

File: swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift
Comment (roop) [reply swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift:69]: This is a very interesting hack, and could be related to the issue I see in the app described in [PR#77 in firezone-apple](https://github.com/firezone/firezone-apple/pull/77#issuecomment-1611006530).

However, I don't want to include this change in this PR. I'll test it out and, if applicable, include it in a later PR.

File: swift/apple/LICENSE
Comment (roop) [reply swift/apple/LICENSE:1]: Fixed in 6e2c29b

File: swift/apple/Makefile
Comment (roop) [reply swift/apple/Makefile]: Fixed in aa5e12a

File: swift/apple/Makefile
Comment (roop) [reply swift/apple/Makefile]: Fixed in aa5e12a (Apparently, I need to do `$(shell name -m)`.)

File: swift/apple/PortalMock/data/jwks.json
Comment (roop) [reply swift/apple/PortalMock/data/jwks.json:1]: Fixed in 75fe3e3

File: swift/apple/PortalMock/data/jwks_priv.json
Comment (roop) [reply swift/apple/PortalMock/data/jwks_priv.json:1]: Fixed in 75fe3e3

File: swift/apple/PortalMock/data/client_csrf_token
Comment (roop) [reply swift/apple/PortalMock/data/client_csrf_token:1]: Fixed in 9a9c9f3

File: swift/apple/LICENSE
Comment (roop) [reply swift/apple/LICENSE]: Fixed in 6e2c29b

**Comment by roop** [comment]
I force-pushed to rebase the commits on top of current cloud.

**Review by jamilbk — APPROVED** [review]

**Review by jamilbk — APPROVED** [review]

**Comment by jamilbk** [comment]
LGTM. Looks like it’s just waiting on a final approval from @conectado or @francesca64 due to the rust workflow change

**Review by conectado — APPROVED** [review]

**Comment by jamilbk** [comment]
Needs final approval from @AndrewDryga and @francesca64

**Review by francesca64 — APPROVED** [review]

**Comment by roop** [comment]
Previously, there was a CI failure at the **Swift / build** step. The failure was in `cargo lipo --release --manifest-path ./Cargo.toml --targets aarch64-apple-darwin,x86_64-apple-darwin` in `./build-rust.sh`, which is invoked by the Xcode run build phase, which is invoked by `./build-framework-dev.sh` when building for macOS.

The errors are weird because it looks as if it's thinking the build is for iOS:

~~~
2023-07-07T12:23:38.3669600Z   = note: ld: warning: building for iOS, but linking in .tbd file (/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/lib/libiconv.tbd) built for macOS/Mac Catalyst
2023-07-07T12:23:38.3697890Z           ld: warning: building for iOS, but linking in .tbd file (/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/System/Library/Frameworks/Security.framework/Security.tbd) built for macOS/Mac Catalyst
2023-07-07T12:23:38.3799530Z           ld: warning: building for iOS, but linking in .tbd file (/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/lib/libc.tbd) built for macOS/Mac Catalyst
2023-07-07T12:23:38.3805290Z           ld: warning: building for iOS, but linking in .tbd file (/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/lib/libSystem.tbd) built for macOS/Mac Catalyst
2023-07-07T12:23:38.3817060Z           ld: warning: building for iOS, but linking in .tbd file (/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/usr/lib/libm.tbd) built for macOS/Mac Catalyst
2023-07-07T12:23:38.3908650Z           ld: building for iOS, but linking in object file built for macOS, file '/Users/runner/work/firezone/firezone/rust/target/aarch64-apple-darwin/release/deps/boringtun-3e0a7e4f6e8ead5b.boringtun.d72eca28-cgu.0.rcgu.o' for architecture arm64
2023-07-07T12:23:38.3918400Z           clang: error: linker command failed with exit code 1 (use -v to see invocation)
2023-07-07T12:23:38.4010350Z           
2023-07-07T12:23:38.4019190Z 
2023-07-07T12:23:38.4111910Z error: could not compile `boringtun` (lib) due to previous error
~~~

So I did a re-run of the job, and it passed then.

**Pull request merged**
